client/state/terms/actions.js

```diff
@@ -16,14 +16,16 @@ import {
  * @param  {Number} siteId   Site ID
  * @param  {String} taxonomy Taxonomy Slug
  * @param  {Array}  terms    An array of term objects
+ * @param  {Object} query    Query Options
  * @return {Object}          Action object
  */
-export function receiveTerms( siteId, taxonomy, terms ) {
+export function receiveTerms( siteId, taxonomy, terms, query = {} ) {
 	return {
 		type: TERMS_RECEIVE,
 		siteId,
 		taxonomy,
-		terms
+		terms,
+		query
```

Comment:
We track queries using the `POSTS_REQUEST_SUCCESS` action type in the posts subtree. I'm not sure I like that approach, but we should think about a consistent approach. I would prefer keeping `POSTS_REQUEST_SUCCESS` limited in its scope (simply to request tracking), but at the same time, I'm not sure "receive" should necessarily be tied to a query*. Here, I think we should either split off a separate action creator/type, or allow `query` to be null (in which case, the `queries` reducer would ignore received items for that action).

_*_ The counter-example being, if I come across a term from some other API endpoint, I should be able to issue an ad-hoc `receiveTerms`.